homeassistant/components/frontend/__init__.py

```diff
@@ -460,6 +460,8 @@ async def get(self, request: web.Request) -> web.Response:
             ),
             content_type="text/html",
         )
+        response.enable_compression()
```

Comment:
Our current index.html is 3300 bytes. Googling gives different answers on if you should compress at such low thresholds or not.

Reply:
It probably only makes sense for a slow 3g connection or bad cell signal.

Reply:
1324 bytes compressed.  Probably 2 packets instead of 4. 🤷

Reply:
For 3G / bad cell it would only be a benefit on the first time the page is loaded. After that the service worker kicks in and preloads the whole frontend. Next time they get index.html, it's served from the local cache while it is being refreshed in the background (in case there was a new frontend):

https://github.com/home-assistant/frontend/blob/dev/src/entrypoints/service_worker.ts#L44-L48

Our authentication requires HTTPS when using a non-internal network domain name, so service worker should always be active when accessing our frontend over a cell phone connection (edge case: VPN to local network).

Reply:
Did some testing with bad signal and network throttling.  No discernible difference with or without compression.

Probably not worth it to change.

Reply:
I did just open a bug fix and an optimization that should bring index.html down from 3300 bytes to 3200 bytes.